drgn/commands/_builtin/crash/_runq.py

```diff
@@ -0,0 +1,192 @@
+from drgn.helpers.linux.sched import task_rq, task_since_last_arrival_ns
+
+
+def get_rq_per_cpu(prog: Program, cpus: List[int] = []) -> Iterator[Tuple[int, Object]]:
+    """
+    Get runqueue for selected cpus
+
+    :param prog: drgn program
+    :param cpus: a list of int
+    :return: Iterator of (int, ``struct rq``) tuples
+    """
+    online_cpus = list(for_each_online_cpu(prog))
+
+    if cpus:
+        selected_cpus = [cpu for cpu in online_cpus if cpu in cpus]
+    else:
+        selected_cpus = online_cpus
+
+    for cpu in selected_cpus:
+        runqueue = per_cpu(prog["runqueues"], cpu)
+        yield (cpu, runqueue)
+
+
+def timestamp_str(ns: int) -> str:
+    """Convert nanoseconds to 'days HH:MM:SS.mmm' string."""
+    ms_total = ns // 1000000
+    secs_total, ms = divmod(ms_total, 1000)
+    mins_total, secs = divmod(secs_total, 60)
+    hours_total, mins = divmod(mins_total, 60)
+    days, hours = divmod(hours_total, 24)
+
+    return f"{days} {hours:02}:{mins:02}:{secs:02}.{ms:03}"
+
+
+@crash_command(
+    description="Display the tasks on the run queues of each cpu.",
+    arguments=(
+        mutually_exclusive_group(
+            argument("-t", action="store_true", dest="show_timestamps"),
+            argument("-T", action="store_true", dest="show_lag"),
+            argument("-m", action="store_true", dest="pretty_runtime"),
+            argument("-g", action="store_true", dest="group"),
+        ),
+        argument("-c", type=str, default="a", dest="cpus"),
+        drgn_argument,
+    ),
+)
+def _crash_cmd_runq(
+    prog: Program, name: str, args: argparse.Namespace, **kwargs: Any
+) -> None:
+    table_format = args.show_timestamps or args.show_lag or args.pretty_runtime
+    table: List[List[Any]] = []
+    headers: List[Any] = []
+
+    runq_clocks: Dict[int, int] = {}
+    cpus = parse_cpuspec(args.cpus).cpus(prog)
+    for i, (cpu, runqueue) in enumerate(get_rq_per_cpu(prog, cpus)):
+        curr_task = runqueue.curr[0].address_of_()
+        curr_task_addr = runqueue.curr.value_()
+        comm = escape_ascii_string(curr_task.comm.string_())
+        pid = curr_task.pid.value_()
+        prio = curr_task.prio.value_()
+        run_time = task_since_last_arrival_ns(curr_task)
+
+        # Show lag (skip formatting if not last CPU)
+        if args.show_lag:
+            runq_clocks[cpu] = task_rq(curr_task).clock.value_()
+            if i == len(cpus) - 1:
+                max_clock = max(runq_clocks.values())
+                lags = {
+                    c: max_clock - runq_clock for c, runq_clock in runq_clocks.items()
+                }
+                sorted_lags = sorted(lags.items(), key=lambda item: item[1])
+                for c, lag in sorted_lags:
+                    print(f"CPU {c}: {lag / 1e9:.2f} secs")
+                return
+            else:
+                continue
+
+        if table_format:
+            row = [
```

Comment:
The format of these options is fairly different from the crash output, was that an intentional choice. That's okay if there's an intentional reason, but it'd be nice to imitate crash's format if not.